packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,223 @@
+import '@aws-cdk/assert/jest';
+import { objectLike } from '@aws-cdk/assert';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import { App, Duration, Lazy, Stack } from '@aws-cdk/core';
+import * as synthetics from '../lib';
+
+let stack: Stack;
+beforeEach(() => {
+  stack = new Stack(new App(), 'canaries');
+});
+
+test('Basic canary properties work', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+    successRetentionPeriod: Duration.days(10),
+    failureRetentionPeriod: Duration.days(10),
+    startAfterCreation: false,
+    timeToLive: Duration.minutes(30),
+    runtime: synthetics.Runtime.SYNTHETICS_1_0,
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    SuccessRetentionPeriod: 10,
+    FailureRetentionPeriod: 10,
+    StartCanaryAfterCreation: false,
+    Schedule: objectLike({ DurationInSeconds: '1800'}),
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Canary can have generated name', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'canariescanary8ee009f',
+  });
+});
+
+test('Name validation does not fail when using Tokens', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: Lazy.stringValue({produce: () => 'myCanary'}),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN: no exception
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary');
+});
+
+test('Throws when name is specified incorrectly', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'myCanary',
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  }))
+    .toThrowError('Canary name must be lowercase, numbers, hyphens, or underscores (no spaces) (myCanary)');
```

Comment:
See how this error reads? It's not ideal...
- Add quotes around the value that failed validation (I typically pass those to `JSON.stringify`)
- The `(no spaces)` seems redundant. Spaces are not lowercase letters, numbers, hyphens or underscores.
- Clearly label the value you received (i.e: `got ${JSON.stringify(value)}`)... Invalid values otherwise may look like a part of the error message...

Reply:
Is there a reason why `JSON.stringify` is necessary here? Or can I just say `got ${value}` if the value itself is a string?